src/plugins/engagement/server/config.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { PluginConfigDescriptor } from 'kibana/server';
+
+const chatConfigSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: false }),
+  chatURL: schema.conditional(
+    schema.siblingRef('enabled'),
+    true,
+    schema.string({ minLength: 1 }),
+    schema.maybe(schema.string())
+  ),
```

Comment:
NIT: I would KISS and remove the conditionals on all those config values. being able to define the properties even when `enabled` is false seems acceptable.

Reply:
Noted.  Chalk this up to not *really* understanding the schema API.  🤷🏻